File: solutions/go/rotational-cipher/1/rotational_cipher.go
package rotationalcipher

import "unicode"

func RotationalCipher(plain string, shiftKey int) string {
    runes := make([]rune, len(plain))
    for i, r := range plain {
        rcode := int(r)
        if unicode.IsLetter(r) {
            rcode += shiftKey 
            switch {
            case 'a' <= r && r <= 'z':
                if (rcode) > int('z') {
                    rcode = rcode - int('z') - 1 + int('a')
                }
			case 'A' <= r && r <= 'Z':
                if (rcode) > int('Z') {
                    rcode = rcode - int('Z') - 1 + int('A')
                }
            }
        } 
        runes[i] = rune(rcode)
    }
    return string(runes)
}

File: solutions/csharp/rotational-cipher/1/RotationalCipher.cs
public static class RotationalCipher
{
    public static string Rotate(string text, int shiftKey)
    {
        if (shiftKey < 0 || shiftKey > 26)
            throw new ArgumentOutOfRangeException(nameof(shiftKey), "Shift key must be between 0 and 26.");

        char[] result = new char[text.Length];
        for (int i = 0; i < text.Length; i++)
        {
            char c = text[i];
            if (char.IsLetter(c))
            {
                char offset = char.IsUpper(c) ? 'A' : 'a';
                result[i] = (char)((((c + shiftKey) - offset) % 26) + offset);
            }
            else
                result[i] = c;
        }
        return new string(result);
    }
}